### PR TITLE
Adding AI metrics for NPM

### PR DESCRIPTION
**Pull request by jaer-tsun**
**What this PR does / why we need it**:
Adding AppInsights metrics for NPM. Pod count, namespace count, and network policy count will be reported on a 30 min heartbeat/interval.

### CODE REVIEW

File: npm/npm.go [hunk 1]
```diff
@@ -77,6 +80,58 @@ func (npMgr *NetworkPolicyManager) GetClusterState() telemetry.ClusterState {
...
+			Name: "NsCount",
+			CustomDimensions: customDimensions,
+		}
+		nwPolicyCount = aitelemetry.Metric{
+			Name: "NwPolicyCount",
+			CustomDimensions: customDimensions,
+		}
+	)
+
+	for ; err != nil; {
```
Comment (tamilmani1989) [inline npm/npm.go]: why are we not checing this error(at 97)  before defining all those metrics?

File: npm/npm.go [hunk 2]
```diff
@@ -77,6 +80,58 @@ func (npMgr *NetworkPolicyManager) GetClusterState() telemetry.ClusterState {
...
+			BatchSize:                 32768,
+			BatchInterval:             30,
+			RefreshTimeout:            15,
+			DebugMode:                 true,
+			GetEnvRetryCount:          5,
+			GetEnvRetryWaitTimeInSecs: 3,
+		}
+		th, err = aitelemetry.NewAITelemetry("", aiMetadata, aiConfig)
+		heartbeat = time.NewTicker(time.Minute * heartbeatIntervalInMinutes).C
+		customDimensions = map[string]string{"ClusterID": util.GetClusterID(npMgr.nodeName),
```
Comment (tamilmani1989) [inline npm/npm.go]: why the assignment was in next line?

File: npm/npm.go [hunk 3]
```diff
@@ -77,6 +80,58 @@ func (npMgr *NetworkPolicyManager) GetClusterState() telemetry.ClusterState {
...
+			AppName:                   util.AzureNpmFlag,
+			AppVersion:                npMgr.version,
+			BatchSize:                 32768,
+			BatchInterval:             30,
+			RefreshTimeout:            15,
+			DebugMode:                 true,
+			GetEnvRetryCount:          5,
+			GetEnvRetryWaitTimeInSecs: 3,
+		}
+		th, err = aitelemetry.NewAITelemetry("", aiMetadata, aiConfig)
```
Comment (tamilmani1989) [inline npm/npm.go]: leave space

File: npm/npm.go [hunk 4]
```diff
@@ -77,6 +80,58 @@ func (npMgr *NetworkPolicyManager) GetClusterState() telemetry.ClusterState {
 	return npMgr.clusterState
 }
 
+// SendAiMetrics :- send NPM metrics using AppInsights
+func (npMgr *NetworkPolicyManager) SendAiMetrics() {
+	var (
+		aiConfig = aitelemetry.AIConfig{
```
Comment (tamilmani1989) [inline npm/npm.go]: i dont think everything has to be in var..just declare fields and define them outside of var..thats usual practice right?

File: npm/npm.go (hunk 1 above)
Comment (jaer-tsun) [reply npm/npm.go]: because that's in a var block

File: npm/npm.go (hunk 2 above)
Comment (jaer-tsun) [reply npm/npm.go]: what do you mean here?

File: npm/npm.go (hunk 4 above)
Comment (jaer-tsun) [reply npm/npm.go]: for init blocks I use var, if it was only a couple or single init I use := || var (name) (type)

File: npm/npm.go (hunk 2 above)
Comment (tamilmani1989) [reply npm/npm.go]: util.GetClusterID(npMgr.nodeName), was in next line, why its not in same line?

File: npm/npm.go (hunk 4 above)
Comment (tamilmani1989) [reply npm/npm.go]: why dont we move outside of var? thats the standard we follow all along.

File: npm/npm.go [hunk 5]
```diff
@@ -77,6 +80,58 @@ func (npMgr *NetworkPolicyManager) GetClusterState() telemetry.ClusterState {
...
+	)
+
+	for ; err != nil; {
+		log.Logf("Failed to init AppInsights with err: %+v", err)
+		time.Sleep(time.Minute * 5)
+		th, err = aitelemetry.NewAITelemetry("", aiMetadata, aiConfig)
+	}
+
+	defer th.Close(10)
+
```
Comment (tamilmani1989) [inline npm/npm.go]: add log saying AI telemetry handle intialized..

**Pull request merged**